src/main/java/org/terasology/launcher/LauncherInitTask.java

```diff
@@ -173,35 +181,48 @@ private BaseLauncherSettings getLauncherSettings(Path launcherDirectory) throws
+     * @param tempDirectory     where to extract the launcher archive for the self-update process
+     * @return true if the self-updater is started for an update, false otherwise
+     */
+    private boolean checkForLauncherUpdates(final Path downloadDirectory, final Path tempDirectory) {
         logger.trace("Check for launcher updates...");
-        boolean selfUpdaterStarted = false;
         updateMessage(BundleUtils.getLabel("splash_launcherUpdateCheck"));
         final LauncherUpdater updater = new LauncherUpdater(TerasologyLauncherVersionInfo.getInstance());
-        if (updater.updateAvailable()) {
+        final Option<GitHubRelease> latestRelease = updater.updateAvailable();
+
+        //TODO: more functional implementation
+        if (latestRelease.isDefined()) {
+            final GitHubRelease release = latestRelease.get();
             logger.trace("Launcher update available!");
             updateMessage(BundleUtils.getLabel("splash_launcherUpdateAvailable"));
-            boolean foundLauncherInstallationDirectory = false;
-            try {
-                updater.detectAndCheckLauncherInstallationDirectory();
-                foundLauncherInstallationDirectory = true;
-            } catch (URISyntaxException | IOException e) {
-                logger.error("The launcher installation directory can not be detected or used!", e);
-                GuiUtils.showErrorMessageDialog(owner, BundleUtils.getLabel("message_error_launcherInstallationDirectory"));
-                // Run launcher without an update. Don't throw a LauncherStartFailedException.
-            }
-            if (foundLauncherInstallationDirectory) {
-                final boolean update = updater.showUpdateDialog(owner);
-                if (update) {
-                    if (saveDownloadedFiles) {
-                        selfUpdaterStarted = updater.update(downloadDirectory, tempDirectory);
-                    } else {
-                        selfUpdaterStarted = updater.update(tempDirectory, tempDirectory);
-                    }
-                }
-            }
+
+            final Try<Path> installationDirectory = updater.detectAndCheckLauncherInstallationDirectory()
+                    .onFailure(e -> {
+                        logger.error("The launcher installation directory can not be detected or used!", e);
+                        GuiUtils.showErrorMessageDialog(owner, BundleUtils.getLabel("message_error_launcherInstallationDirectory"));
+                    });
+
+            installationDirectory
+                    //TODO: .filter(directory -> updater.showUpdateDialog(owner, directory, release))
+                    .map(directory -> updater.showUpdateDialog(owner, directory, release))
+                    //TODO: .map(directory -> updater.update(downloadDirectory, tempDirectory, directory, release))
+                    //TODO: .getOrElse(false);
+                    .onSuccess(this::showDownloadPage);
```

Comment:
This is where we attempt to open the download page instead of running the self update for now.